ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -142,15 +226,25 @@ long getDefaultMaxOutstandingRequestBytes() {
     }
 
     static OutputType get(Env env) {
-      if (env.containsKey(OUTPUT_BUCKET)) {
+      boolean hasBigQueryOutputMode = env.containsKey(BIG_QUERY_OUTPUT_MODE);
+      if (env.containsKey(OUTPUT_BUCKET) && !hasBigQueryOutputMode) {
         return OutputType.gcs;
-      } else if (env.containsKey(OUTPUT_TOPIC)) {
+      } else if (env.containsKey(OUTPUT_TOPIC) && !hasBigQueryOutputMode) {
         return OutputType.pubsub;
-      } else if (env.containsKey(OUTPUT_TABLE)) {
-        return OutputType.bigQuery;
+      } else if (env.containsKey(OUTPUT_TABLE) || hasBigQueryOutputMode) {
+        final String outputMode = env.getString(BIG_QUERY_OUTPUT_MODE, "streaming").toLowerCase();
+        if (outputMode.equals("streaming")) {
```

Comment:
Because we're immediately returning or throwing in all cases, this could likely be an elegant switch statement.